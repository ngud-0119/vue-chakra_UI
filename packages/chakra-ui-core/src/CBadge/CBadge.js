/**
 * Hey! Welcome to @chakra-ui/vue Badge
 *
 * The Badge component is used to highlight an item's status for quick recognition.
 *
 * @see Docs     https://vue.chakra-ui.com/badge
 * @see Source   https://github.com/chakra-ui/chakra-ui-vue/blob/master/packages/chakra-ui-core/src/CBadge/CBadge.js
 */

import { forwardProps } from '../utils'
import { baseProps } from '../config/props'
import CBox from '../CBox'
import useBadgeStyles from './utils/badge.styles'

/**
 * CBadge component
 *
 * Used to highlight an item's status for quick recognition.
 *
 * @extends CBox
 * @see Docs https://vue.chakra-ui.com/badge
 */
const CBadge = {
  name: 'CBadge',
  inject: ['$chakraTheme', '$chakraColorMode'],
  props: {
    variant: {
      type: String,
      default: 'subtle'
    },
    variantColor: {
      type: String,
      default: 'gray'
    },
    ...baseProps
  },
  computed: {
    colorMode () {
      return this.$chakraColorMode()
    }
  },
  render (h) {
    const badgeStyleProps = useBadgeStyles({
      theme: this.$chakraTheme(),
      colorMode: this.colorMode,
      color: this.variantColor,
      variant: this.variant
    })

    return h(CBox, {
      props: {
        d: 'inline-block',
        textTransform: 'uppercase',
        fontSize: 'xs',
        fontFamily: 'body',
        px: 1,
        rounded: 'sm',
        fontWeight: 'bold',
        whiteSpace: 'nowrap',
        verticalAlign: 'middle',
        ...badgeStyleProps,
        ...forwardProps(this.$props)
      },
      attrs: {
        'data-chakra-component': 'CBadge'
      }
    }, this.$slots.default)
  }
}

export default CBadge
