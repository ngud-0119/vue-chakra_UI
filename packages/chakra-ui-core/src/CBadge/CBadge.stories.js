import { storiesOf } from '@storybook/vue'
import { CBadge } from '..'

storiesOf('UI | Badge', module)
  .add('Default Badge', () => ({
    components: { CBadge },
    template: `
      <div>
        <CBadge>Default</CBadge>
      </div>
    `
  }))
  .add('With color', () => ({
    components: { CBadge },
    template: `
      <div>
        <CBadge mx="2">Default</CBadge>
        <CBadge mx="2" variant-color="green">Success</CBadge>
        <CBadge mx="2" variant-color="red">Removed</CBadge>
        <CBadge mx="2" variant-color="indigo">New</CBadge>
      </div>
    `
  }))
