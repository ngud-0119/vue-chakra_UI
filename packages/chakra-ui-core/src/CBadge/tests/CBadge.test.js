import CBadge from '../'
import { render } from '@/tests/test-utils'

const renderComponent = (props) => {
  const base = {
    components: { CBadge },
    template: `<CBadge>500</CBadge>`,
    ...props
  }
  return render(base)
}

it('should render correctly', () => {
  const { asFragment } = renderComponent()
  expect(asFragment()).toMatchSnapshot()
})

it('should display children', () => {
  const { getByText } = renderComponent()

  expect(getByText('500')).toBeInTheDocument()
})
