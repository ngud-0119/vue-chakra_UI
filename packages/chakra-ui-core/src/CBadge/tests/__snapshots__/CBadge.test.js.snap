// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should apply variant styles corectly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  display: inline-block;
  text-transform: uppercase;
  font-size: var(--chakra-fontSizes-xs);
  font-family: var(--chakra-fonts-body);
  -webkit-padding-start: var(--chakra-space-1);
  padding-inline-start: var(--chakra-space-1);
  -webkit-padding-end: var(--chakra-space-1);
  padding-inline-end: var(--chakra-space-1);
  border-radius: var(--chakra-radii-sm);
  font-weight: var(--chakra-fontWeights-bold);
  white-space: nowrap;
  vertical-align: middle;
  background-color: var(--chakra-colors-green-100);
  color: var(--chakra-colors-green-800);
  -webkit-margin-start: var(--chakra-space-2);
  margin-inline-start: var(--chakra-space-2);
  -webkit-margin-end: var(--chakra-space-2);
  margin-inline-end: var(--chakra-space-2);
  margin-right: var(--chakra-space-2);
}

.emotion-3 {
  display: inline-block;
  text-transform: uppercase;
  font-size: var(--chakra-fontSizes-xs);
  font-family: var(--chakra-fonts-body);
  -webkit-padding-start: var(--chakra-space-1);
  padding-inline-start: var(--chakra-space-1);
  -webkit-padding-end: var(--chakra-space-1);
  padding-inline-end: var(--chakra-space-1);
  border-radius: var(--chakra-radii-sm);
  font-weight: var(--chakra-fontWeights-bold);
  white-space: nowrap;
  vertical-align: middle;
  background-color: var(--chakra-colors-green-500);
  color: var(--chakra-colors-white);
  -webkit-margin-start: var(--chakra-space-2);
  margin-inline-start: var(--chakra-space-2);
  -webkit-margin-end: var(--chakra-space-2);
  margin-inline-end: var(--chakra-space-2);
  margin-right: var(--chakra-space-2);
}

.emotion-4 {
  display: inline-block;
  text-transform: uppercase;
  font-size: var(--chakra-fontSizes-xs);
  font-family: var(--chakra-fonts-body);
  -webkit-padding-start: var(--chakra-space-1);
  padding-inline-start: var(--chakra-space-1);
  -webkit-padding-end: var(--chakra-space-1);
  padding-inline-end: var(--chakra-space-1);
  border-radius: var(--chakra-radii-sm);
  font-weight: var(--chakra-fontWeights-bold);
  white-space: nowrap;
  vertical-align: middle;
  box-shadow: inset 0 0 0px 1px #3ea76a;
  color: var(--chakra-colors-green-500);
  -webkit-margin-start: var(--chakra-space-2);
  margin-inline-start: var(--chakra-space-2);
  -webkit-margin-end: var(--chakra-space-2);
  margin-inline-end: var(--chakra-space-2);
}

<div
    class="emotion-0 emotion-1"
    data-chakra-component="CStack"
  >
    <div
      class="emotion-2"
      data-chakra-component="CBadge"
    >
      Subtle
    </div>
    <div
      class="emotion-3"
      data-chakra-component="CBadge"
    >
      Solid
    </div>
    <div
      class="emotion-4"
      data-chakra-component="CBadge"
    >
      Outline
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: inline-block;
  text-transform: uppercase;
  font-size: var(--chakra-fontSizes-xs);
  font-family: var(--chakra-fonts-body);
  -webkit-padding-start: var(--chakra-space-1);
  padding-inline-start: var(--chakra-space-1);
  -webkit-padding-end: var(--chakra-space-1);
  padding-inline-end: var(--chakra-space-1);
  border-radius: var(--chakra-radii-sm);
  font-weight: var(--chakra-fontWeights-bold);
  white-space: nowrap;
  vertical-align: middle;
  background-color: var(--chakra-colors-gray-100);
  color: var(--chakra-colors-gray-800);
}

<div
    class="emotion-0"
    data-chakra-component="CBadge"
  >
    500
  </div>
</DocumentFragment>
`;
